akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/VersionVector.scala

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.internal
+import scala.annotation.tailrec
+import scala.collection.immutable.TreeMap
+import akka.annotation.InternalApi
+
+/**
+ * VersionVector module with helper classes and methods.
+ */
+object VersionVector {
+
+  private val emptyVersions: TreeMap[String, Long] = TreeMap.empty
+  val empty: VersionVector = ManyVersionVector(emptyVersions)
+
+  def apply(): VersionVector = empty
+
+  def apply(versions: TreeMap[String, Long]): VersionVector =
+    if (versions.isEmpty) empty
+    else if (versions.size == 1) apply(versions.head._1, versions.head._2)
+    else ManyVersionVector(versions)
+
+  def apply(key: String, version: Long): VersionVector = OneVersionVector(key, version)
+
+  /** INTERNAL API */
+  @InternalApi private[akka] def apply(versions: List[(String, Long)]): VersionVector =
+    if (versions.isEmpty) empty
+    else if (versions.tail.isEmpty) apply(versions.head._1, versions.head._2)
+    else apply(emptyVersions ++ versions)
+
+  /**
+   * Java API
+   */
+  def create(): VersionVector = empty
```

Comment:
Since it is internal we can skip the Java API and convenience constructors we do not use ourselves?

Reply:
👍